src/main/webapp/go.js

```diff
@@ -48,10 +66,87 @@ function getHunt() {
       puzzleArr.push(mssg.items[i].riddle.puzzle);
       destArr.push(mssg.items[i].name);
     }
+    createMap();
     updateToCurrentState(destIndex);
   });
 }
 
+/**
+ * Add a marker to the map at the specified location.
+ * @param {Double} destLat: latitude of location.
+ * @param {Double} destLng: longitude of location.
+ * @param {String} destName: Name of location.
+ */
+function addMarkerToMap(destLat, destLng, destName) {
+  const coord = new google.maps.LatLng(destLat, destLng);
+  const marker = new google.maps.Marker({
+    position: coord,
+    title: destName,
+  });
+  marker.setMap(map);
+}
+
+/**
+ * Creates a map and adds it to the page.
+ */
+function createMap() {
+  map = new google.maps.Map(
+      document.getElementById(MAP_DISPLAY),
+      // Centered at GooglePlex (updated below).
+      {center: {lat: 37.422, lng: -122.084}, zoom: 7},
+  );
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      const pos = {lat: position.coords.latitude,
+        lng: position.coords.longitude};
+      addMarkerToMap(position.coords.latitude, position.coords.longitude,
+          'Your current location');
+      map.setCenter(pos);
+    }, function() {
+      updateMessage(MAP_MSSG_DISPLAY, 'Error: I can\'t find your location.');
+    });
```

Comment:
This is more of a suggestion than a directive, but this might be a good opportunity to use [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) here!  Arrow functions are nice because they produce shorter code, and also handle the `this` binding differently.  If you're not familiar with javascript scope and the `this` keyword, you can read [this article](https://medium.com/tfogo/advantages-and-pitfalls-of-arrow-functions-a16f0835799e#:~:text=Arrow%20functions%20are%20best%20for,of%20this%20in%20their%20scope.).  Let me know if you run into any issues.

```suggestion
    navigator.geolocation.getCurrentPosition((position) => {
      const pos = {lat: position.coords.latitude,
        lng: position.coords.longitude};
      addMarkerToMap(position.coords.latitude, position.coords.longitude,
          'Your current location');
      map.setCenter(pos);
    }, 
    () => updateMessage(MAP_MSSG_DISPLAY, 'Error: I can\'t find your location.'));
```